### ISSUE TITLE
implementing notes - Note.finishEdit() doesn't toggle editing state if App.editNote() isn't defined

### ISSUE THREAD

**Issue by jvanbrug** [survivejs/webpack_react#353]
The following doesn't set `editing` to `false` if `this.props.onEdit` is undefined:

```javascript
finishEdit = (e) => {
  const value = e.target.value;
  if(this.props.onEdit && value.trim()) {
    this.props.onEdit(value);
    this.setState({
      editing: false
    });
  }
};
```

So `this.setState()` just needs to move outside the `if`:

```javascript
finishEdit = (e) => {
  const value = e.target.value;
  if(this.props.onEdit && value.trim()) {
    this.props.onEdit(value);
  }
  this.setState({
    editing: false
  });
};
```

`this.props.onEdit` is set in the next section, when `App.editNode()` is defined and passed to `Note` through `Notes`. However, before the next section, the reader is prompted to try editing Notes:

>If you try to edit a Note now, you should see an input and be able to edit the data. Given we haven't set up onEdit handler, it doesn't do anything useful yet, though. We'll need to capture the edited data next and update App state so that the code works.

At this point in the tutorial, notes can go into editing mode, but they can't leave editing mode if the user presses `Enter` or blurs (clicks outside the note).

**Comment by jvanbrug** [comment]
I'm at this point in the book, so I don't yet understand how this issue should be resolved in future chapters.

**Comment by bebraw** [comment]
Yeah, that's the expected behavior. I added a snippet like this for rc4:

```
If you try to edit a `Note` now, you should see an input and be able to edit the data. Given we haven't set up `onEdit` handler, it doesn't do anything useful yet, though. We'll need to capture the edited data next and update `App` state so that the code works.
```

Dealing with `onEdit` is the next step after that as it takes some further thinking.

**Comment by jvanbrug** [comment]
Shouldn't `this.state.editable` always be `false` after calling `finishEdit`, even if `onEdit` is undefined or `value` is empty/whitespace?

**Comment by bebraw** [comment]
> Shouldn't this.state.editable always be false after calling finishEdit, even if onEdit is undefined or value is empty/whitespace?

It's inside `if` because I don't want to allow empty input to go through. But doing that could be an alternative.

I recall the main reason for leaving it inside `if` had to do with the edit logic. If you an empty value, it's not possible to trigger edit at all in the current setup.

**Comment by jvanbrug** [comment]
I see. I find it more intuitive that after the following actions:

* clicking task
* typing empty input
* clicking away from task

would result in the note returning to its previous, "viewing" mode with the old task displayed, rather than staying in "editing" mode with an empty task.

**Comment by bebraw** [comment]
Yeah, that would be a nice touch UX-wise. Thanks for the idea!

I'll see if I can get this tweak into the final release. :+1:

**Comment by jvanbrug** [comment]
:+1: happy to adjust/add commits or submit a PR

**Comment by bebraw** [comment]
Implemented at 9190a9d. Forgot to add # to the commit message so no auto-closing...
